lib/mongo/cluster/sdam_flow.rb

```diff
@@ -356,7 +368,7 @@ def commit_changes
       # (which is important for spec tests because they have real i/o
       # happening against bogus addresses which yield unknown responses
       # before the responses are mocked).
-      unless updated_desc.unknown? && previous_desc.unknown?
+      unless updated_desc.unknown? && previous_desc.unknown? || updated_desc.object_id == previous_desc.object_id
```

Comment:
Using `unless` along with both an `&&` and an `||` makes this a bit complex to think through (especially if the reader doesn't remember which of `&&` and `||` has higher precedence; can we refactor this to be a bit easier to read? I think something like this would be easier for people to read:

```ruby
if updated_desc.unknown? && previous_desc.unknown? 
  return
end

if updated_desc.object_id == previous_desc.object_id
  return
end

publish_sdam_event(...) # etc.
```